app/main/viewmanager.ts

```diff
@@ -0,0 +1,161 @@
+'use strict';
+
+import { ipcMain, BrowserView } from 'electron';
+import { View, ViewProps } from './view';
+import { mainWindow } from '.';
+
+import ConfigUtil = require('../renderer/js/utils/config-util');
+
+// ViewManager Class allows us to create, delete and manage several instances of View class
+class ViewManager {
+	views: { [key: number]: View };
+	selectedIndex: number;
+
+	constructor() {
+		this.views = {};
+		this.selectedIndex = -1;
+		this.registerIpcs();
+	}
+
+	registerIpcs(): void {
+		ipcMain.on('create-view', (e: Event, props: ViewProps) => {
+			this.create(props);
+		});
+
+		ipcMain.on('select-view', (e: Event, index: number) => {
+			this.select(index);
+		});
+
+		ipcMain.on('destroy-view', (e: Event, index: number) => {
+			this.destroy(index);
+		});
+
+		ipcMain.on('destroy-all-views', () => {
+			this.destroyAll();
+		});
+
+		ipcMain.on('forward-view-message', (e: Event, name: string, ...params: any[]) => {
+			this.views[this.selectedIndex].webContents.send(name, ...params);
+		});
+
+		ipcMain.on('forward-message-all', (e: Event, name: string, ...params: any[]) => {
+			this.forwardMessageAll(name, ...params);
+		});
+
+		ipcMain.on('call-view-function', (e: Event, name: string, ...params: any[]) => {
+			this.callViewFunction(this.selectedIndex, name, ...params);
+		});
+
+		ipcMain.on('call-specific-view-function', (e: Event, index: number, name: string, ...params: any[]) => {
```

Comment:
Please document this newly added IPCs -
> forward-message-all
> call-view-function
> call-specific-view-function
> forward-view-message

@priyank-p happy with the naming of ipc functions?

Reply:
Documented these.

Reply:
I think we can rename `forward-view-message` to `forward-message-view` and `forward-message-all` to `forward-message-all-views`. Similarly the `forwardMessageAll` to `forwardMessageAllViews`. @akashnimare thoughts?